woocommerce/woocommerce
# Add offline payment methods screen

### Changes proposed in this Pull Request:

Partially addresses #52178

Adds offline payment gateways screen. Note that:

1. This is based on a work in progress, thus the design will not be exactly as shown in [Figma](https://github.com/user-attachments/assets/00a6c65c-5d34-4cb8-8fc2-ac7ea3e4cea6). There's a [draft PR](https://github.com/woocommerce/woocommerce/pull/52161) that's addressing design discrepancies, and follow-up PRs will be needed to address offline gateways designs.  
2. Currently there is no react routing between the main payment page and offline gateway page, but we should address that in follow-ups.
3. Sorting of gateways is not addressed in this PR.

### How to test the changes in this Pull Request:

1. Ensure all offline gateways are disabled
1. Enable `reactify-classic-payments-settings` feature flag
4. Go to `WooCommerce > Settings > Payments`
6. Click on the chevron icon on `Offline payment methods`
7. Observe there's "Direct bank transfer", "Check payments", "Cash on delivery" gateways displayed as per screenshot
8. Click on `Enable` on each offline gateway and observe it shows loading state and changes to `Manage` button
9. Refresh the page and observe the button persists
10. Click on `Manage` on any of the gateways
11. Observe you're redirected to the gateway configuration screen
12. Disable the gateway and save
13. Go back to the offline gateways screen and observe the button is changed back to `Enable`

<img width="1237" alt="image" src="https://github.com/user-attachments/assets/f47d4677-c8ed-454c-9093-40a5b8b57b07">



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
